src/common/transform.h

```diff
@@ -125,19 +125,22 @@ class Transform {
 
       GPUSet devices = distribution_.Devices();
       size_t range_size = *range_.end() - *range_.begin();
+
+      // Extract index to deal with possible old OpenMP.
+      size_t device_beg = *(devices.begin());
+      size_t device_end = *(devices.end());
 #pragma omp parallel for schedule(static, 1) if (devices.Size() > 1)
-      for (omp_ulong i = 0; i < devices.Size(); ++i) {
-        int d = devices.Index(i);
+      for (omp_ulong device = device_beg; device < device_end; ++device) {  // NOLINT
         // Ignore other attributes of GPUDistribution for spliting index.
+        int d_unnormalised = devices.Index(device);
```

Comment:
In situations like this you need to have some index beginning from 0 you could get it from the enclosing for loop:
```
 for (omp_ulong i = 0; i < devices.Size(); ++i) {
    auto device_idx = devices[i];
    size_t shard_size =
            GPUDistribution::Block(devices).ShardSize(range_size, i);
```